scripts/generate_refcats_gen3.py

```diff
@@ -0,0 +1,248 @@
+
+
+########################################
+# Identify all required shards
+
+HTM_LEVEL = 7
+
+
+def _get_shards(centers, radius):
+    """Return all shards overlapping a set of fields.
+
+    Parameters
+    ----------
+    centers : iterable [`astropy.coordinates.SkyCoord`]
+        The right ascension and declination of the field centers.
+    radius : `float`
+        The radius of each field, in degrees.
+
+    Returns
+    -------
+    shards : `set` [`int`]
+        The set of HTM indices that overlap any of the fields.
+    """
+    indexer = htm.HTM(depth=HTM_LEVEL)
+    shards = set()
+    for center in centers:
+        shards.update(indexer.intersect(center.ra.degree, center.dec.degree, radius, inclusive=True))
+    return shards
+
+
+logging.info("Identifying refcat shards...")
+shards = _get_shards(FIELDS, FIELD_RADIUS)
+if not shards:
+    raise RuntimeError("No HTM shards found; coordinates are likely corrupted.")
+logging.debug("%d shards found", len(shards))
+
+
+# Simplify shard list so that Butler doesn't choke on a long query string.
+
+def _make_range(start, end):
+    """Represent a range of contiguous integers in Butler dimension
+    expression syntax.
+
+    Parameters
+    ----------
+    start, end : `int`
+        The first and last elements of the range, *inclusive*.
+        Assumes ``start <= end``.
+
+    Returns
+    -------
+    range : `str`
+        The Butler syntax for the range.
+    """
+    if start == end:
+        return str(start)
+    else:
+        return f"{start}..{end}"
+
+
+def _to_ranges(ids):
+    """Reexpress a set of integer IDs in terms of Butler dimension
+    expression ranges.
+
+    Parameters
+    ----------
+    ids : iterable [`int`]
+        The IDs to be compressed. Must not contain duplicates.
+
+    Returns
+    -------
+    ranges : iterable [`str`]
+        A collection of Butler range literals and integer literals equivalent
+        to ``ids``. See
+        https://pipelines.lsst.io/modules/lsst.daf.butler/queries.html#literals
+        for syntax.
+
+    Examples
+    --------
+    >>> _to_ranges([1, 2, 3, 5])
+    ["1..3", "5"]
+    """
+    ranges = []
+
+    asc_ids = sorted(ids)
+    range_start = asc_ids[0]
+    # Can replace with itertools.pairwise in Python 3.10
+    for last_id, id in zip(asc_ids, asc_ids[1:]):
+        # Loop invariant: last_id is an element of the contiguous range
+        #     starting with range_start
+        # Loop invariant: range_start <= last_id < id
+        if id > last_id + 1:
+            # last_id ends the range, and id starts a new one
+            ranges.append(_make_range(range_start, last_id))
+            range_start = id
+    ranges.append(_make_range(range_start, id))  # id is last element in asc_ids
+    return ranges
+
+
+id_ranges = _to_ranges(shards)
+
+
+########################################
+# Transfer shards
+
+DEST_DIR = "${AP_VERIFY_HITS2015_DIR}/preloaded/"
+STD_REFCAT = "refcats"
+DEST_RUN = "refcats/imported"
```

Comment:
Do we not want this to just be `refcats`, since it's going into the test repo?

Reply:
This is more flexible, especially when I'm trying to test [DM-29857](https://jira.lsstcorp.org/browse/DM-29857) and during the transition period. Using `refcats` here would require clobbering all existing (known good) refcats.

Reply:
So long as that gets chained back to "refcats", I think we're fine? Or do we not even need that?

Reply:
I was planning to chain it, yes. (Current format of `preloaded/` is that `refcats` is a chain containing `refcats/gen2`.)